# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

i have choosen the content_refresh_anonymized lane
As the data contains the detailed info of webpage performance , including search impressions ,clicks , ranking position , content age and more ...
i selected this as found this data very realistic , as many websites often contain this

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision : which webpage to review first for content refresh based on there search performance and engagement matrics?
unit of analysis : each row represents a single webpage
output: ranked list of webpages that should be prioritized for content and refresh .
action : action will be taken by the seo specialist team and the content team to decide which page to be updated and improve user performace and usage
cost of wrong recommendation : if wrong page is prioritized then important pages may losse there visibility and traffic while other will get that importance, This reduces the efficiency of content optimization efforts.
why ml : doing this all manually is tough and ml will help us to find hidden patterns and insights in the data , like clicks , impressions , CTR and more of the features in data . ML help to prioritize the webpage using these factores , rather than doing it mannually


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

It appears the dataset `content_refresh_anonymized.csv` is hosted in a GitHub repository. We will download it from its raw URL.

In [17]:

csv_url = "https://raw.githubusercontent.com/Nikhilsankhyan0/flyrank-ml-internship/refs/heads/main/data/raw/content_refresh_anonymized.csv"

# # Using !wget to download the file directly into the Colab environment
# !wget -O /content/content_refresh_anonymized.csv {csv_url}

print("File downloaded to /content/content_refresh_anonymized.csv")

File downloaded to /content/content_refresh_anonymized.csv


In [18]:
import pandas as pd


df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)

print("\nTrend Direction")
print(df["trend_direction"].value_counts())

print("\nContent Types")
print(df["content_type"].value_counts())

print("\nAverage Metrics")
print(df[["impressions_90d",
          "clicks_90d",
          "ctr",
          "avg_position",
          "content_age_days"]].describe())

Dataset Shape: (30000, 44)

Trend Direction
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Content Types
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Average Metrics
       impressions_90d    clicks_90d           ctr  avg_position  \
count     30000.000000  30000.000000  30000.000000   30000.00000   
mean       5200.366300     16.097333      0.510733      16.34238   
std       16838.019547     75.076958      3.279162      15.21679   
min           1.000000      0.000000      0.000000       0.00000   
25%          81.000000      0.000000      0.000000       6.20000   
50%         731.000000      1.000000      0.070000      10.80000   
75%        3615.250000      7.000000      0.290000      22.30000   
max      517715.000000   4178.000000    100.000000     245.00000   

       content_age_days  
count       30000.00000  
mean          

## Observations

The starter dataset contains **30,000 webpages** and **44 features**, providing sufficient information to analyze search performance, content quality, and user engagement.

Among the 30,000 pages, **16,262 pages (approximately 54%) are classified as "down"**, making declining content the largest group in the dataset. This indicates that identifying pages requiring content refresh is a meaningful and practical problem.

The dataset is dominated by **keyword articles (27,207 pages, about 91%)**, while Feedly articles (2,096) and comparison articles (697) make up a much smaller portion. This suggests that the analysis will primarily focus on keyword-driven content.

Search visibility varies considerably across pages. The average page receives **5,200 impressions** over 90 days, but the median is only **731 impressions**, indicating a highly skewed distribution where a small number of pages receive significantly higher traffic than most others.

The average content age is **256 days**, with most pages ranging between **132 and 333 days** old. This makes content freshness an important factor to investigate when prioritizing pages for review.

These observations support the **Refresh / Content Opportunity Scoring** lane because the dataset contains a large number of declining pages, substantial variation in search visibility, and multiple content quality signals that can be combined to prioritize pages for content review.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What my work say :**
This project provides **decision support** by identifying webpages that may require content review based on observable search and engagement signals.
The findings are **observational and directional**, helping prioritize pages for review rather than making guaranteed predictions.

**What my work can't say** :
This project **cannot prove** that refreshing a webpage will directly improve its rankings or traffic, since correlation does not establish causation.
It does **not predict Google's ranking algorithm** or claim to explain how Google ranks webpages.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.